src/main/java/ttt/gui/GuiGameController.java

```diff
@@ -5,53 +5,85 @@
 
 import java.util.List;
 
+import static ttt.player.GuiHumanPlayer.IGNORE_AS_MOVE_WILL_COME_FROM_DISPLAY;
+
 public class GuiGameController implements GameController {
 
+    private final GameConfiguration gameConfiguration;
     private GameRules ticTacToeRules;
     private DisplayPresenter boardView;
+    private GameType gameType;
 
-    public GuiGameController(GameRules ticTacToeRules, ViewFactory viewFactory) {
+    public GuiGameController(GameConfiguration gameConfiguration, GameRules ticTacToeRules, ViewFactory viewFactory) {
+        this.gameConfiguration = gameConfiguration;
         this.ticTacToeRules = ticTacToeRules;
         this.boardView = viewFactory.createView(this, ticTacToeRules);
     }
 
     @Override
     public void presentGameTypes() {
-        List<GameType> allGameTypes = ticTacToeRules.getGameTypes();
+        List<GameType> allGameTypes = gameConfiguration.getGameTypes();
         boardView.presentGameTypes(allGameTypes);
     }
 
     @Override
     public void presentBoardDimensionsFor(GameType gameType) {
-        ticTacToeRules.storeGameType(gameType);
-        String dimension = ticTacToeRules.getDimension(gameType);
+        setGameType(gameType);
+        String dimension = gameConfiguration.getDimension(gameType);
         boardView.presentGridDimensionsUpTo(dimension);
     }
 
     @Override
     public void presentBoard(String dimension) {
-        ticTacToeRules.initialiseGame(dimension);
+        ticTacToeRules.initialiseGame(gameType, dimension);
         Board board = ticTacToeRules.getBoard();
+
+        playAutomatedMoveIfAppropriate();
         boardView.presentsBoard(board);
```

Comment:
when the board is drawn, if the game type is 'Unbeatable vs Human', then the unbeatable player needs to have taken its turn. The first attempt had the code

 `if (gameType == UNBEATABLE_VS_HUMAN) {
        int automatedMove = ticTacToeRules.getCurrentPlayersNextMove();
        playMove(automatedMove); 
   }` 

but this violates the open close principle. Therefore, I introduced a 'GuiHumanPlayer' which essentially is just a place holder so that the code can be treated a bit more generically. The game can ask the player for the move. If the move returned is '-1' (which is the constant IGNORE_AS_MOVE_WILL_COME_FROM_DISPLAY), then the game can ignore it, as it is a human player, and their input will come from the next button clicked.